include/fuse.h

```diff
@@ -1004,6 +1004,17 @@ int fuse_getgroups(int size, gid_t list[]);
 int fuse_interrupted(void);
 
 /**
+ * Invalidates cache for the given path.
+ *
+ * This calls fuse_lowlevel_notify_inval_inode internally.
+ * This only does something for nodes which are present in
+ * the hashes; it is a no-op for paths to forgotten nodes.
+ *
+ * @return 0 on success, negative error value otherwise.
```

Comment:
I do not have an example of what this information would be useful for either. And I concede your point about the call becoming more complicated.

However, I am concerned about the time when someone comes up with a use-case that would benefit from having this information. At that point, the API cannot be changed without breaking backwards compatibility. Therefore, I would like to see the ENOENT returned as-is.
People who would like to ignore this error can then trivially write a wrapper function. If we did not return ENOENT, there would be no recourse for people who want it.

From that point of view, reducing the return value to just `(1,-1,0)` would be even worse.